### PR TITLE
feat(EMS-772-773-776): Eligibility - already have an account page/flow. Update required data middleware checks

### PR DESCRIPTION
**Pull request by ttbarnes**
This PR adds the "Do you already have an account" page to the end of the insurance eligibility flow. Depending on the submitted yes/no answer, we redirect to new (currently empty routes) of "create an account" or "sign in".

## Changes
- Add a temporary "create account" button and route for the "eligible to apply online" page, redirecting to the "create account page". This is purely for development and means that we can:
  - Continue building the rest of the insurance flow without being blocked/hindered by account creation.
  - Progress account creation.
  - This route/button will be removed once we have account creation and sign in fully working.
- Create "already have an account" field ID, route, controller and page.
- Add E2E tests for submitting yes and no in the new page.
- Update the fields used for the "initial checks" task list. We are now filtering out the `ALREADY_HAVE_ACCOUNT` field. This field is part of the eligibility flow, but is useless information to save. Without saving or stripping this out, the eligibility task will have a value of "in progress", which is incorrect.
- Update the "get application" middleware so that we only get an application for specific routes (`RELEVANT_ROUTES`).
- Add "create account" and "sign in" routes to the list of routes that do _not_ require data checking, in the `requiredInsuranceEligibilityDataProvided` middleware.

## Other improvements

- Add/align root constants for all insurance sections.
- Add destructuring to some routes used in the E2E tests fora "eligible to apply online" page.
- Simplify some URL assertions in E2E tests.

### CODE REVIEW

File: src/ui/server/helpers/task-list/generate-groups-and-tasks/initial-checks.ts [hunk 1]
```diff
@@ -13,7 +15,7 @@ const createInitialChecksTasks = (): Array<TaskListDataTask> => [
     href: '#',
     title: INITIAL_CHECKS.TASKS.ELIGIBILITY,
     id: TASK_IDS.INITIAL_CHECKS.ELIGIBILITY,
-    fields: Object.values(FIELD_IDS.INSURANCE.ELIGIBILITY),
+    fields: Object.values(FIELD_IDS.INSURANCE.ELIGIBILITY).filter((fieldId) => fieldId !== ALREADY_HAVE_ACCOUNT),
```
Comment (Zainzzkk) [inline src/ui/server/helpers/task-list/generate-groups-and-tasks/initial-checks.ts:18]: a comment here for why please

File: src/ui/server/middleware/insurance/get-application/index.ts [hunk 2]
```diff
@@ -1,11 +1,27 @@
...
+ * If the route is a relevant insurance route, get the application and add to res.locals
+ * @param {Express.Request} Express request
+ * @param {Express.Response} Express response
+ * @returns {Express.Request.Next} request next or response redirect
+ */
 const getApplicationMiddleware = async (req: Request, res: Response, next: Next) => {
   const { originalUrl: url } = req;
 
-  if (url.includes(ROUTES.INSURANCE.ELIGIBILITY_ROOT)) {
+  if (!RELEVANT_ROUTES.some((route) => url.includes(route))) {
```
Comment (Zainzzkk) [inline src/ui/server/middleware/insurance/get-application/index.ts:24]: can this be explained with a comment please

File: src/ui/templates/insurance/eligibility/eligible-to-apply-online.njk [hunk 3]
```diff
@@ -45,6 +45,16 @@
       }
     }) }}
 
+    {# temporary POST button whilst we develop account creation #}
+    {{ govukButton({
+      text: "Create an account",
+      classes: "govuk-button--secondary",
+      attributes: {
+        formaction: '/insurance/eligibility/eligible-to-apply-online/temp-create',
```
Comment (Zainzzkk) [inline src/ui/templates/insurance/eligibility/eligible-to-apply-online.njk:53]: can this be passed as a variable?

File: src/ui/server/controllers/insurance/eligibility/already-have-account/index.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+export const TEMPLATE = TEMPLATES.INSURANCE.ELIGIBILITY.ALREADY_HAVE_ACCOUNT;
+
+/**
+ * get
+ * Render the Do you already have an account page
+ * @param {Express.Request} Express request
+ * @param {Express.Response} Express response
+ * @returns {Express.Response.render} Do you already have an account page
+ */
+export const get = (req: Request, res: Response) => res.render(TEMPLATE, singleInputPageVariables({ ...PAGE_VARIABLES, BACK_LINK: req.headers.referer }));
```
[the remark this answers is not in the document]
Comment (ttbarnes) [reply src/ui/server/controllers/insurance/eligibility/already-have-account/index.ts:28]: No 😢 i'm with you though it's more readable on a new line

File: src/ui/templates/insurance/eligibility/eligible-to-apply-online.njk (hunk 3 above)
Comment (ttbarnes) [reply src/ui/templates/insurance/eligibility/eligible-to-apply-online.njk:53]: I did consider this, I think it's overkill as it's a temporary thing and will be removed entirely when we have account creation and sign in working?

**Review by Zainzzkk — APPROVED** [review]
LGTM

**Pull request merged**